carma-messenger-ui/website/widgets/emergencyResponse/widget.js

```diff
@@ -0,0 +1,328 @@
+/***
+* Create a unique namespace for each plugin widget to minimize collision of same name variables or functions.
+***/
+CarmaJS.registerNamespace("CarmaJS.WidgetFramework.emergencyResponse");
+var listenerAlert;
+var listenerBSM;
+
+const UNAVAILABLE_SPEED = 8191;
+const SPEED50TH = 0.02;
+//enumeration values for siren_in_use:
+const siren_in_use = {
+    UNAVAILABLE: 0,
+    NOT_IN_USE: 1,
+    IN_USE: 2,
+    RESERVED: 3
+}
+
+
+// enumeration values for lightbar_in_use:
+const lightbar_in_use = {
+    UNAVAILABLE: 0,
+    NOT_IN_USE: 1,
+    IN_USE: 2,
+    YELLOW_CAUTION_LIGHTS: 3,
+    SCHOOL_BUS_LIGHTS: 4,
+    ARROW_SIGNS_ACTIVE: 5,
+    SLOW_MOVING_VEHICLE: 6,
+    FREQ_STOPS: 7
+}
+
+//Display vehicle information
+var subscribe_bsm = () => {
+    listenerBSM = new ROSLIB.Topic({
+        ros: ros,
+        name: '/bsm_outbound',
+        messageType: 'cav_msgs/BSM'
+    });
+
+    listenerBSM.subscribe(function (message) {
+        if (message.core_data != undefined && message.core_data.latitude != undefined && message.core_data.longitude != undefined) {
+            $("#positionValue").text(message.core_data.latitude + "," + message.core_data.longitude);
+        }
+
+        if (message.core_data != undefined && message.core_data.id != undefined) {
+            if (message.core_data.id.length != 0) {
+                $("#bsmIdValue").text(message.core_data.id);
+            }
+        }
+
+        if (message.core_data != undefined && message.core_data.speed != undefined) {
+            if (message.core_data.speed != UNAVAILABLE_SPEED) {
+                let speedMPH = message.core_data.speed / SPEED50TH;
```

Comment:
The cav_msgs (carma_v2x_msgs) BSM message includes message.core_data.speed in units m/s, so the conversion with SPEED50TH is not needed here. Instead, a conversion can be used to convert from m/s to mph.